packages/components/menu/src/useMenu.ts

```diff
@@ -109,18 +109,30 @@ export function useMenu({
     prevOpenRef.current = isOpen;
   }, [isOpen]);
 
-  const handleOpenChange = React.useCallback(
-    (nextOpen: boolean) => {
+  const handleOpenChange = React.useCallback<
+    UseFloatingOptions['onOpenChange']
+  >(
+    (nextOpen, _event, reason) => {
+      // if the user made component controlled by providing isOpen prop
+      // but onOpen callback is not provided, we won't add toggle logic
+      // to the trigger component. So they can make any toggle logic on their own.
+      const isFullyControlled = isControlled && !onOpen;
```

Comment:
We currently have a type guard implemented that would forbid this mixed state from happening in the first place.